### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by grafgeest**
[task3] Unit testing
image

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -20,16 +28,17 @@ describe('AppComponent', () => {
...
-    expect(app.title).toEqual('angular2019q3');
-  });
-
-  it('should render title', () => {
-    const fixture = TestBed.createComponent(AppComponent);
-    fixture.detectChanges();
-    const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('.content span').textContent).toContain('angular2019q3 app is running!');
-  });
+  // TODO remove
```
Comment (vnovikov-clgx) [inline src/app/app.component.spec.ts]: why you can't remove it in this pull?

File: src/app/core/components/course-page/course-page.component.spec.ts [hunk 2]
```diff
@@ -16,10 +30,37 @@ describe('CoursePageComponent', () => {
...
+    pageComponent = fixture.nativeElement;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should render course page', () => {
+    expect(pageComponent.querySelectorAll('courses-page')).toBeDefined();
```
Comment (vnovikov-clgx) [inline src/app/core/components/course-page/course-page.component.spec.ts]: if `querySelectorAll` will return an empty array, this test will pass?

File: src/app/core/components/course-page/course-page.component.spec.ts [hunk 3]
```diff
@@ -16,10 +30,37 @@ describe('CoursePageComponent', () => {
...
+    spyOn(component.edited, 'emit');
+
+    pageComponent.querySelector<HTMLButtonElement>('.edit').click();
+
+    expect(component.edited.emit).toHaveBeenCalledWith(courseMock);
+    expect(consoleSpy).toHaveBeenCalledWith('onEdit is clicked');
+  });
+
+  it('should call delete when clicked edit button', () => {
+    const consoleSpy = spyOn(console, 'log');
```
Comment (vnovikov-clgx) [inline src/app/core/components/course-page/course-page.component.spec.ts]: You can create this spy in the `beforeEach` method. You do not need to copy this  into each test

File: src/app/shared/header/header.component.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <div class="app-header">
   <div class="app-header-container container">
     <app-logo class="mr-auto"></app-logo>
-    <div class="button "><span class="icon icon-user"></span>{{login}}</div>
-    <div class="button "><span class="icon icon-exit"></span>Log off</div>
+    <div class="button log-in"><span class="icon icon-user"></span>{{login}}</div>
+    <div class="button log-off"><span class="icon icon-exit"></span>Log off</div>
```
Comment (vnovikov-clgx) [inline src/app/shared/header/header.component.html]: minor: mb `log-out`?

File: src/styles.scss [hunk 5]
```diff
@@ -1,4 +1,5 @@
 /* You can add global styles to this file, and also import other style files */
+//@import "~bootstrap/dist/css/bootstrap.min.css";
```
Comment (vnovikov-clgx) [inline src/styles.scss]: remove comments

File: src/app/core/components/course-list/course-list.component.spec.ts [hunk 6]
```diff
@@ -1,25 +1,38 @@
...
+import {CourseService} from '../../services/course.service';
+import {SearchControlComponent} from '../search-control/search-control.component';
+import {CoursePageComponent} from '../course-page/course-page.component';
+import {FormsModule} from '@angular/forms';
+import { MOCK_COURSES } from '../../services/mock-courses';
 
 describe('CourseListComponent', () => {
   let component: CourseListComponent;
   let fixture: ComponentFixture<CourseListComponent>;
+  let service: CourseService;
```
Comment (vnovikov-clgx) [inline src/app/core/components/course-list/course-list.component.spec.ts]: You should mock this service, how you will test what was return?

File: src/styles.scss (hunk 5 above)
Comment (grafgeest) [reply src/styles.scss]: Done

File: src/app/app.component.spec.ts (hunk 1 above)
Comment (grafgeest) [reply src/app/app.component.spec.ts]: Done

File: src/app/core/components/course-page/course-page.component.spec.ts (hunk 3 above)
Comment (grafgeest) [reply src/app/core/components/course-page/course-page.component.spec.ts]: Done

File: src/app/core/components/course-page/course-page.component.spec.ts (hunk 2 above)
Comment (grafgeest) [reply src/app/core/components/course-page/course-page.component.spec.ts]: Done

File: src/app/core/components/course-list/course-list.component.spec.ts (hunk 6 above)
Comment (grafgeest) [reply src/app/core/components/course-list/course-list.component.spec.ts]: Done, I used spyOn.

**Pull request merged**